tests/conformance/test_conformance.py

```diff
@@ -0,0 +1,712 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import requests
+import tempfile
+import uuid
+import logging
+import functools
+import pytest
+
+from google.cloud import storage
+
+from . import _read_local_json
+
+
+_CONFORMANCE_TESTS = _read_local_json("retry_strategy_test_data.json")[
+    "retryStrategyTests"
+]
+
+_STORAGE_EMULATOR_ENV_VAR = "STORAGE_EMULATOR_HOST"
+"""Environment variable defining host for Storage emulator."""
+
+_CONF_TEST_PROJECT_ID = "my-project-id"
+_CONF_TEST_SERVICE_ACCOUNT_EMAIL = (
+    "my-service-account@my-project-id.iam.gserviceaccount.com"
+)
+
+_STRING_CONTENT = "hello world"
+_BYTE_CONTENT = b"12345678"
+
+
+########################################################################################################################################
+### Library methods for mapping ########################################################################################################
+########################################################################################################################################
+
+
+def bucket_get_blob(client, _preconditions, **resources):
+    bucket = resources.get("bucket")
```

Comment:
How do these fail if a resource is missing for some reason?

Reply:
The resources are populated through pytest fixtures. If for any reason, a resource is missing, we expect to either see a Fixture Error or an AttributeError. 

If a fixture has a problem, pytest will stop executing fixtures for that test and mark the test as having an error. Another case would be raising an AttributeError when attempting to retrieve an attribute of the resource (most cases retrieving `bucket.name`)